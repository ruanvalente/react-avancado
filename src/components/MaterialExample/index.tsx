import { Button } from '@mui/material'

import * as S from './styles'

const MaterialExample = () => (
  <S.Wrapper>
    <Button variant="contained" color="primary">
      MaterialExample
    </Button>
  </S.Wrapper>
)

export default MaterialExample
