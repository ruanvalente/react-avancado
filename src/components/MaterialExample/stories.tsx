import { Story, Meta } from '@storybook/react'
import MaterialExample from '.'

export default {
  title: 'MaterialExample',
  component: MaterialExample
} as Meta

export const Default: Story = () => <MaterialExample />
