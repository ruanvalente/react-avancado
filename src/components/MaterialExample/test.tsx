import { render, screen } from '@testing-library/react'

import MaterialExample from '.'

describe('<MaterialExample />', () => {
  it('should render the heading', () => {
    const { container } = render(<MaterialExample />)

    expect(
      screen.getByRole('button', { name: /MaterialExample/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
