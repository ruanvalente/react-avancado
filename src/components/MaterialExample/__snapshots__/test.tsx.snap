// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaterialExample /> should render the heading 1`] = `
<main
  class=""
>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    MaterialExample
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</main>
`
